py-polars/tests/unit/io/test_pyarrow_dataset.py

```diff
@@ -199,3 +202,34 @@ def test_pyarrow_dataset_comm_subplan_elim(tmp_path: Path) -> None:
     assert lf0.join(lf1, on="a", how="inner").collect().to_dict(as_series=False) == {
         "a": [1, 2]
     }
+
+
+def test_pyarrow_dataset_with_pyarrow_options(tmp_path: Path) -> None:
+    df = pl.DataFrame({"a": [1, 2, 3]})
+
+    file_path_0 = tmp_path / "0.parquet"
+
+    df.write_parquet(file_path_0)
+
+    pa_ds = ds.dataset(file_path_0, format="parquet")
+    lf = pl.scan_pyarrow_dataset(pa_ds, pyarrow_options={"use_threads": False})
+    assert lf.collect().to_dict(as_series=False) == {"a": [1, 2, 3]}
+
+
+def test_pyarrow_dataset_with_conflicting_pyarrow_options(tmp_path: Path) -> None:
+    df = pl.DataFrame({"a": [1, 2, 3]})
+
+    file_path_0 = tmp_path / "0.parquet"
+
+    df.write_parquet(file_path_0)
+
+    pa_ds = ds.dataset(file_path_0, format="parquet")
+    with pytest.raises(pl.ComputeError) as exc_info:
+        pl.scan_pyarrow_dataset(
+            pa_ds, batch_size=200, pyarrow_options={"batch_size": 500}
+        ).collect()
```

Comment:
Why is this a ComputeError? I thought you raised a ValueError in your code? Is it because it goes through the Rust side somehow?